jobs/cloud_replicate.Jenkinsfile

```diff
@@ -0,0 +1,147 @@
+            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+            cosa init --branch ${ref} ${pipecfg.source_config.url}
+            cosa buildfetch --build=${params.VERSION} \
+                --arch=all --url=s3://${s3_stream_dir}/builds \
+                --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG}
+            """)
+        }
+
+        def builtarches = shwrapCapture("""
+                          cosa shell -- cat builds/builds.json | \
+                              jq -r '.builds | map(select(.id == \"${params.VERSION}\"))[].arches[]'
+                          """).split() as Set
+        assert builtarches.contains("x86_64"): "The x86_64 architecture was not in builtarches."
+        if (!builtarches.containsAll(basearches)) {
+            if (params.ALLOW_MISSING_ARCHES) {
+                echo "Some requested architectures did not successfully build! Continuing."
+                basearches = builtarches.intersect(basearches)
+            } else {
+                echo "ERROR: Some requested architectures did not successfully build"
+                echo "ERROR: Detected built architectures: $builtarches"
+                echo "ERROR: Requested base architectures: $basearches"
+                currentBuild.result = 'FAILURE'
+                return
+            }
+        }
+
+        // Update description based on updated set of architectures
+        build_description += "[${params.STREAM}][${basearches.join(' ')}][${params.VERSION}]"
+        currentBuild.description = "${build_description} Running"
+
+        for (basearch in basearches) {
+            libcloud.replicate_to_clouds(pipecfg, basearch, params.VERSION)
+        }
+
+        stage('Publish') {
+            pipeutils.withAWSBuildUploadCredentials() {
+                // Since some of the earlier operations (like AWS replication) only modify
+                // the individual meta.json files we need to re-generate the release metadata
+                // to get the new info and upload it back to s3.
+                def arch_args = basearches.collect{"--arch ${it}"}.join(" ")
+                shwrap("""
+                cosa generate-release-meta --build-id ${params.VERSION} --workdir .
+                cosa buildupload --build=${params.VERSION} --skip-builds-json \
+                    ${arch_args} s3 --aws-config-file=\${AWS_BUILD_UPLOAD_CONFIG} \
+                    --acl=public-read ${s3_stream_dir}/builds
+                """)
+
+            }
+
+        }
+
+        currentBuild.result = 'SUCCESS'
+        currentBuild.description = "${build_description} ✓"
+
+// main try finishes here
+} catch (e) {
+    currentBuild.result = 'FAILURE'
+    throw e
+} finally {
+    if (currentBuild.result != 'SUCCESS') {
+        pipeutils.trySlackSend(color: 'danger', message: ":fcos: :bullettrain_front: :trashfire: cloud-replicate <${env.BUILD_URL}|#${env.BUILD_NUMBER}> [${params.STREAM}][${basearches.join(' ')}] (${params.VERSION})")
```

Comment:
Let's use a different emoji from the release job one? Maybe e.g. `:cloud:`?

Reply:
updated to match what jenkins prints for the stage name for the cloud replication.